app/models/request.rb

```diff
@@ -54,6 +56,10 @@ def total_items
     request_items.sum { |item| item["quantity"] }
   end
 
+  def total_items_fromstr
```

Comment:
Can we rename this? `fromstr` doesn't seem to mean anything. Actually... why do we need this and `total_items`? It seems like `total_items` just has a flat out bug because it's trying to sum strings (?)

Reply:
`request_items` attribute of `Request` model is a `jsonb` column on the `requests` table. It contains values such as:

```json
[{"item_id": 32, "quantity": 3}, {"item_id": 15, "quantity": 35}]
```

i.e., when saved in the database, the quantities are all integers.

However, I think when the form is initially submitted, they could be strings. This [model test](https://github.com/rubyforgood/human-essentials/blob/main/spec/models/request_spec.rb#L35) verifies that the string quantities are coerced into integers before saving.

The coercion logic is implemented as a before_save callback in the `Request` model:
```ruby
class Request < ApplicationRecord
  before_save :sanitize_items_data

  def sanitize_items_data
    return unless request_items && request_items_changed?

    self.request_items = request_items.map do |item|
      item.merge("item_id" => item["item_id"]&.to_i, "quantity" => item["quantity"]&.to_i)
    end
  end
end
```

For the confirmation modal logic, we intentionally do not save the request model, it is only initialized, therefore, this coercion logic has not yet run. So to achieve the sum logic required for partner quota warning, requires converting potential string values to integers.

This may be fixed by swapping out the `before_save` for a `before_validation` callback, but requires some investigation to see if this actually works, and if it doesn't break any existing logic. I'll look into this further...